examples/user_guide/Linking_Plots.ipynb

```diff
@@ -0,0 +1,220 @@
+    "\n",
+    "One of the simplest examples of a Link is the DataLink which links the data from two sources as long as they match in length, e.g. below we create two elements with data of the same length. By declaring a ``DataLink`` between the two we can ensure they are linked and can be selected together:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from holoviews.plotting.links import DataLink\n",
+    "\n",
+    "scatter1 = hv.Scatter(np.random.randn(1000, 2))\n",
+    "scatter2 = hv.Scatter(np.random.randn(1000, 2)*2, 'x2', 'y2')\n",
+    "\n",
+    "DataLink(scatter1, scatter2)\n",
+    "\n",
+    "(scatter1 + scatter2).options('Scatter', tools=['box_select'], clone=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Writing a Link\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A ``Link`` consists of two components the ``Link`` itself and a ``LinkCallback`` which provides the actual implementation behind the ``Link``. In order to demonstrate writing a ``Link`` we'll start with a fairly straightforward example, linking an ``HLine`` to the mean y-value of a selection on a ``Scatter`` element. To express this we declare a ``MeanHLineLink`` class subclassing from the ``Link`` baseclass and declare ``ClassSelector`` parameters with the appropriate types to perform some basic validation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import param\n",
+    "from holoviews.plotting.links import Link\n",
+    "\n",
+    "class MeanHLineLink(Link):\n",
+    "    \n",
+    "    source = param.ClassSelector(class_=hv.Scatter)\n",
+    "    target = param.ClassSelector(class_=hv.HLine)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we have the ``Link`` class we need to write the implementation in the form of a ``LinkCallback``. A ``LinkCallback`` should declare the ``source_model`` we want to listen to events on and a ``target_model``, declaring which model should be altered in response. To find out which models we can attach the ``Link`` to we can create a ``Plot`` instance and look at the ``plot.handles``, e.g. here we create a ``ScatterPlot`` and can see it has a 'source', which represents the ``ColumnDataSource``."
```

Comment:
Right after '... write the implementation in the form of a ``LinkCallback``' I would point to the two types of bokeh callback with the link (html this time!) to the bokeh docs. This helps make it more concrete by linking to bokeh earlier.